airflow_dbt_python/hooks/dbt.py

```diff
@@ -555,22 +599,101 @@ class DbtSnowflakeHook(DbtHook):
     conn_type = "snowflake"
     hook_name = "dbt Snowflake Hook"
     conn_params = [
-        DbtConnectionParam("conn_type", "type", "postgres"),
+        DbtConnectionParam("conn_type", "type", conn_type),
         "host",
         "schema",
-        DbtConnectionParam("login", "user"),
-        "password",
+        DbtConnectionParam(
+            "login",
+            "user",
+            depends_on=lambda x: x.extra_dejson.get("authenticator", "") != "oauth",
+        ),
+        DbtConnectionParam(
+            "login",
+            "oauth_client_id",
+            depends_on=lambda x: x.extra_dejson.get("authenticator", "") == "oauth",
+        ),
+        DbtConnectionParam(
+            "password",
+            depends_on=lambda x: not any(
+                (
+                    *(k in x.extra_dejson for k in ("private_key_file", "private_key_content")),
```

Comment:
I don't think we care whether there are private key file and contents in `extra_dejson` as long as we have our password and `.get("authenticator", "") == "oauth"`.

Reply:
I think we should keep the mental model small for users with only one key to consider to determine what everything else means (authenticator).